src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -786,24 +609,8 @@ public static float L2DistSquared(ReadOnlySpan<float> left, ReadOnlySpan<float>
             Contracts.Assert(count <= left.Length);
             Contracts.Assert(count <= right.Length);
 
-            if (count < MinInputSize || !Sse.IsSupported)
-            {
-                float norm = 0;
-                for (int i = 0; i < count; i++)
-                {
-                    float distance = left[i] - right[i];
-                    norm += distance * distance;
-                }
-                return norm;
-            }
-            else if (Avx.IsSupported)
-            {
-                return AvxIntrinsics.Dist2(left, right, count);
-            }
-            else
-            {
-                return SseIntrinsics.Dist2(left, right, count);
-            }
+            var value = TensorPrimitives.Distance(left.Slice(0, count), right.Slice(0, count));
```

Comment:
```suggestion
            float value = TensorPrimitives.Distance(left.Slice(0, count), right.Slice(0, count));
```

Reply:
Currently our editor config has us prefer `var`. If that is not what we want then I'll want to update the editor config as well to reflect that change.

Reply:
The previous code was using int and float in these methods.

Reply:
> Currently our editor config has us prefer var

dotnet/runtime notably has `use var only when apparent` and this matches what the compiler part of dotnet/roslyn does.

Some code bases do what you've got in ML.NET, which is `use var everywhere`. This is for example what the IDE portion of dotnet/roslyn does.

There isn't, strictly speaking, a right or wrong answer. Both sides have good reasons on why having the type visible or hidden should be sufficient and it ultimately comes down to preference (I use var everywhere in my own personal projects).

-- Using `var` only when apparent makes it very clear what the type is, without needing to use an IDE or IntelliSense. It can also avoid accidental bugs on copy/paste and other factors.

-- On the other hand, one can argue that the type is largely not important to the algorithm. Functional languages often rely heavily on type inference for example, it can make it easier to share code, and the exact type might not always exist (such as for generics, chained methods, and so on). There are also many cases where the return type might be considered "apparent" to the developer, but where it isn't considered as such by the analyzer rules (for example, `MyType.Create(...)` is not apparent, nor is `x.ToString()`).